libnavigation-core/src/test/java/com/mapbox/navigation/core/reroute/MapboxRerouteControllerTest.kt

```diff
@@ -0,0 +1,333 @@
+package com.mapbox.navigation.core.reroute
+
+import com.mapbox.api.directions.v5.models.RouteOptions
+import com.mapbox.base.common.logger.Logger
+import com.mapbox.navigation.core.directions.session.DirectionsSession
+import com.mapbox.navigation.core.directions.session.RoutesRequestCallback
+import com.mapbox.navigation.core.routeoptions.RouteOptionsProvider
+import com.mapbox.navigation.core.trip.session.TripSession
+import io.mockk.MockKAnnotations
+import io.mockk.Ordering
+import io.mockk.every
+import io.mockk.impl.annotations.MockK
+import io.mockk.mockk
+import io.mockk.slot
+import io.mockk.spyk
+import io.mockk.verify
+import io.mockk.verifyOrder
+import org.junit.After
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+
+class MapboxRerouteControllerTest {
+
+    private lateinit var rerouteController: MapboxRerouteController
+
+    @MockK
+    private lateinit var directionsSession: DirectionsSession
+
+    @MockK
+    private lateinit var tripSession: TripSession
+
+    @MockK
+    private lateinit var routeOptionsProvider: RouteOptionsProvider
+
+    @MockK
+    private lateinit var logger: Logger
+
+    @MockK
+    private lateinit var routeOptions: RouteOptions
+
+    @MockK
+    private lateinit var routeCallback: RerouteController.RoutesCallback
+
+    @MockK
+    lateinit var primaryRerouteObserver: RerouteController.RerouteStateObserver
+
+    @Before
+    fun setup() {
+        MockKAnnotations.init(this, relaxUnitFun = true, relaxed = true)
+        rerouteController = spyk(
+            MapboxRerouteController(
+                directionsSession,
+                tripSession,
+                routeOptionsProvider,
+                logger
+            )
+        )
+    }
```

Comment:
nit: can be a preference, but this does the same thing

```suggestion
    private val directionsSession: DirectionsSession = mockk(relaxed = true)
    private val tripSession: TripSession = mockk(relaxed = true)
    private val routeOptionsProvider: RouteOptionsProvider = mockk()
    private val logger: Logger = mockk(relaxed = true)
    private val routeOptions: RouteOptions = mockk()
    private val routeCallback: RerouteController.RoutesCallback = mockk()
    private val primaryRerouteObserver: RerouteController.RerouteStateObserver = mockk(relaxed = true)

    private val rerouteController: MapboxRerouteController = spyk(
        MapboxRerouteController(
            directionsSession,
            tripSession,
            routeOptionsProvider,
            logger
        )
    )
```

Reply:
it's almost the same, the only difference that annotated properties are created for each test and don't hold any state